service/src/main/java/greencity/security/service/AuthorityServiceImpl.java

```diff
@@ -17,9 +17,55 @@ public class AuthorityServiceImpl implements AuthorityService {
     private final UserRepo userRepo;
 
     @Override
-    public Set<String> getAllEmploeesAuthorities(String email) {
+    public Set<String> getAllEmployeesAuthorities(String email) {
         User user =
             userRepo.findByEmail(email).orElseThrow(() -> new NotFoundException(ErrorMessage.USER_NOT_FOUND_BY_EMAIL));
         return authorityRepo.getAuthoritiesByEmployeeId(user.getId());
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void updateEmployeesAuthorities(UserEmployeeAuthorityDto dto, String email) {
+        User user =
+            userRepo.findByEmail(email)
+                .orElseThrow(() -> new NotFoundException(ErrorMessage.USER_NOT_FOUND_BY_EMAIL + email));
+        User employee =
+            userRepo.findById(dto.getEmployeeId())
+                .orElseThrow(() -> new NotFoundException(ErrorMessage.USER_NOT_FOUND_BY_ID + dto.getEmployeeId()));
+        if (!user.getRole().equals(Role.ROLE_ADMIN) || employee.getRole().equals(Role.ROLE_USER)
+            || user.getEmail().equals(employee.getEmail())) {
+            throw new BadRequestException(ErrorMessage.USER_HAS_NO_PERMISSION);
+        }
+
+        List<Authority> authorities = employee.getAuthorities();
+        for (String name : dto.getAuthorities()) {
+            Authority authority = authorityRepo.findByName(name)
+                .orElseThrow(() -> new NotFoundException(ErrorMessage.NOT_FOUND_AUTHORITY + name));
+            if (Boolean.TRUE.equals(checkAuthoritiesEmployee(authorities, authority))) {
```

Comment:
Why do you use wrapper class here and in checkAuthoritiesEmployee()? It seems to me that returned value cannot be null by method logic